erts/preloaded/src/prim_file.erl

```diff
@@ -598,20 +601,37 @@ read_link_info(Name, Opts) ->
 read_info_1(Name, FollowLinks, TimeType) ->
     try
         case read_info_nif(encode_path(Name), FollowLinks) of
-            {error, Reason} ->
-                {error, Reason};
-            FileInfo ->
-                CTime = from_posix_seconds(FileInfo#file_info.ctime, TimeType),
-                MTime = from_posix_seconds(FileInfo#file_info.mtime, TimeType),
-                ATime = from_posix_seconds(FileInfo#file_info.atime, TimeType),
-                {ok, FileInfo#file_info{ ctime = CTime,
-                                         mtime = MTime,
-                                         atime = ATime }}
+            {error, Reason} -> {error, Reason};
+            FileInfo -> {ok, adjust_times(FileInfo, TimeType)}
+        end
+    catch
+        error:_ -> {error, badarg}
+    end.
+
+read_handle_info(Fd) ->
+  read_handle_info_1(Fd, local).
+read_handle_info(Fd, Opts) ->
+  read_handle_info_1(Fd, proplist_get_value(time, Opts, local)).
+
+read_handle_info_1(Fd, TimeType) ->
+    try
+        #{ handle := FRef } = get_fd_data(Fd),
+        case read_handle_info_nif(FRef) of
+            {error, Reason} -> {error, Reason};
+            FileInfo -> {ok, adjust_times(FileInfo, TimeType)}
         end
     catch
         error:_ -> {error, badarg}
     end.
 
+adjust_times(FileInfo, TimeType) ->
```

Comment:
Would it make sense to catch the `posix` `TimeType` here and skip all the conversion logic and copying of the `file_info` record?

Reply:
Probably, yes, but since the original code doesn't do that I didn't want to make unrelated changes.